File: app/src/main/java/ru/mamykin/foreignbooksreader/views/ReadBookView.java
package ru.mamykin.foreignbooksreader.views;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface ReadBookView extends MvpView {
    void showToast(String text);

    void showLoading(boolean loading);

    void setBookName(String name);

    void setReadPages(String text);

    void setReadPercent(String text);

    void setSourceText(CharSequence text);

    void setTranslationText(String text);

    void showTranslationPopup(String original, String translation);

    void initBookView(String title, String text);

    void showBookContent(boolean show);
}

File: app/src/main/java/ru/mamykin/foboreader/presentation/readbook/ReadBookView.kt
package ru.mamykin.foboreader.presentation.readbook

import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import ru.mamykin.foboreader.core.ui.BaseView

@StateStrategyType(AddToEndSingleStrategy::class)
interface ReadBookView : BaseView {

    fun showParagraphLoading(show: Boolean)

    fun showParagraphTranslation(text: String)

    fun showWordLoading(show: Boolean)

    fun showWordTranslation(word: String, translation: String)

    fun showBookName(name: String)
}